Fails on Ubuntu 12.04.2 LTS.

```
./test_bson
        d : 1    3.140000
        s : 2    hello
        s_n : 2          goodbye
        o : 3    
                a : 4    
                        0 : 5    BSON_BINDATA


        u : 6    BSON_UNDEFINED
        oid : 7          010203040506070809101112
        b : 8    true
        date : 9         72623859790382856
        n : 10   BSON_NULL
        r : 11   BSON_REGEX: ^asdf
        c : 13   BSON_CODE: function(){}
        c_n : 13         BSON_CODE: function(){}
        symbol : 14      SYMBOL: symbol
        symbol_n : 14    SYMBOL: symbol
        cws : 15         BSON_CODE_W_SCOPE: function(){return i}
         SCOPE:         i : 16   123

        timestamp : 17   i: 1, t: 2
        l : 18   1234605616436508552

sizeof(time_t) is 8


Failed ASSERT [test/bson_test.c] (343):
     bson_oid_generated_time(&oid) < cur_time + 1

./test_bson failed!
make: *** [test] Error 1
```

```
ubuntu:~/mongo-c-driver$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=12.04
DISTRIB_CODENAME=precise
DISTRIB_DESCRIPTION="Ubuntu 12.04.2 LTS"
```

--- a/src/bson.c
+++ b/src/bson.c
@@ -177,7 +177,7 @@ MONGO_EXPORT void bson_oid_gen( bson_oid_t *oid ) {
 }
 
 MONGO_EXPORT time_t bson_oid_generated_time( bson_oid_t *oid ) {
-    int out = 0;
+    time_t out = 0;
     bson_big_endian32( &out, &oid->ints[0] );
 
     return out;

